packages/framework/ai-collab/README.md

```diff
@@ -1,34 +1,194 @@
+		metadata: { description: `The list of task groups that are being managed by this task management application.` },
+	}),
+}) {}
+```
+
+### Example 1: Collaborate with AI
+```ts
+import { aiCollab } from "@fluid-experimental/ai-collab";
+import { PlannerAppState } from "./types.ts"
+// This is not a real file, this is meant to represent how you initialize your app data.
+import { initializeAppState } from "./yourAppInitializationFile.ts"
+
+//  --------- File name: "app.ts" ---------
+
+// Initialize your app state somehow
+const appState: PlannerAppState = initializeAppState({
+		taskGroups: [
+		{
+			title: "My First Task Group",
+			description: "Placeholder for first task group",
+			tasks: [
+				{
+					assignee: "Alice",
+					title: "Task #1",
+					description:
+						"This is the first Sample task.",
+					priority: "low",
+					complexity: 1,
+					status: "todo",
+				},
+			],
+			engineers: [
+				{
+					name: "Alice",
+					maxCapacity: 15,
+					skills:
+						"Senior engineer capable of handling complex tasks. Versed in most languages",
+				},
+				{
+					name: "Charlie",
+					maxCapacity: 7,
+					skills: "Junior engineer capable of handling simple tasks. Versed in Node.JS",
+				},
+			],
+		},
+	],
+})
+
+// Typically, the user would input this through a UI form/input of some sort.
+const userAsk = "Update the task group description to be a about creating a new Todo list application. Create a set of tasks to accomplish this and assign them to the available engineers. Keep in mind the max capacity of each engineer as you assign tasks."
+
+// Collaborate with AI one function call.
+const response = await aiCollab<typeof PlannerAppState>({
+		openAI: {
+			client: new OpenAI({
+				apiKey: OPENAI_API_KEY,
+			}),
+			modelName: "gpt-4o",
+		},
+		treeView: view,
+		treeNode: view.root.taskGroups[0],
+		prompt: {
+			systemRoleContext:
+				"You are a manager that is helping out with a project management tool. You have been asked to edit a group of tasks.",
+			userAsk: userAsk,
+		},
+		planningStep: true,
+		finalReviewStep: true,
+		dumpDebugLog: true,
+	});
+
+if (response.status === 'sucess') {
+	// Render the UI view of your task groups.
+	window.alert(`The AI has successfully completed your request.`);
+} else {
+	window.alert(`Something went wrong! response status: ${response.status}, error message: ${response.errorMessage}`);
+}
+
+
+```
+
+Once the `aiCollab` function call is initiated, an LLM will immediately begin attempting to make changes to your Shared Tree using the provided user prompt, the types of your SharedTree and the provided app guidance. The LLM produces multiple changes, in a loop asynchronously. Meaning, you will immediatley see changes if your UI's render loop is connected to your SharedTree App State.
+
+### Example 2: Collaborate with AI onto a branched state and let the user merge the review and merge the branch back manually
+- **Coming Soon**
```

Comment:
TODO before merging the PR? Or for later?

```suggestion

- **Coming Soon**
```

Reply:
Will add it back after, I need to look at the newly added branching API changes and figure out how they're used. Will come back with a work item.